### ISSUE TITLE
Use `sizeof` when appropriate

### ISSUE THREAD

**Issue by op2786** [Vector35/binaryninja-api#3931]
```
140002dbf  struct rc4_ctx rc
140002dbf  memset(&rc, 0, 0x100)
```

`rc4_ctx` is 256 byte long structure, so BN can use `sizeof(rc)` instead of `0x100` in order to help me understand that all structure will be zeroed in this case.

**Comment by plafosse** [comment]
This is a really good idea. I think the way to go about this is to make it part of the IntegerDisplayType system. Very much like how the enum dialog works. You tell it to display as a `sizeof` and then you have to select what type. By default it might be able to look up the types that are in scope and select the size type whose size matches most closely. This would also allow you to change any integer into a `sizeof` operation. Hmm makes me think we should do `offsetof` as well 🤔

**Comment by plafosse** [comment]
If the selected type doesn't match the integer value then it can still display it in terms of the selected variable's type.
```
memset(&var4 + 0x10, 0, sizeof(var4) - 0x10)
```
Now as to the age old question should it be `sizeof(<type>)` or `sizeof(<variable>)` my vote is for `sizeof(<variable>)` Unless there is no variable of that type in the current function.

**Comment by pjcreath** [comment]
Yes, please!  IDA uses "T" for this and filters your options to expressions (`sizeof`, `offsetof`, etc.) that match the value, much like "M" for enums.

For items that don't match the integer value, IDA provides a custom offset expression tool, where you can specify a base value with an adjustment, after which the base value (which then matches) could be associated with a sizeof/offsetof.

I'm totally fine with this being a manual annotation.

**Comment by utkonos** [comment]
This would be very useful indeed. Right now I use a comment wherever these occur.

<img width="592" height="172" alt="Image" src="https://github.com/user-attachments/assets/d0015a4b-4935-4568-9b77-2d59a999a3c9" />

<img width="419" height="116" alt="Image" src="https://github.com/user-attachments/assets/d3a0f47a-42b8-4165-8091-94bfd0e00a0b" />

If there were a way to actually annotate these using a display as choice somehow, that would be ideal.

**Comment by utkonos** [comment]
@plafosse Would this one, #3931, be handled the same way as #3530 ?

**Comment by plafosse** [comment]
Nope completely different analysis is needed.... Actually this could really easily be done with an IntegerDisplayType 🤔
